addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1334,25 +1492,75 @@ var VideoWidget = MediaWidget.extend({
      * @param {Object} options
      * @returns {Object}
      *          $video -> the created video jQuery node
-     *          type -> the type of the created video
+     *          platform -> the type of the created video
      *          errorCode -> if defined, either '0' for invalid URL or '1' for
      *              unsupported video provider
      */
     _createVideoNode: async function (url, options) {
         options = options || {};
-        const videoData = await this._getVideoURLData(url, options);
-        if (videoData.error) {
-            return {errorCode: 0};
-        }
-        if (!videoData.platform) {
-            return {errorCode: 1};
-        }
-        const $video = $('<iframe>').width(1280).height(720)
-            .attr('frameborder', 0)
-            .attr('src', videoData.embed_url)
-            .addClass('o_video_dialog_iframe');
+        if (options['uploaded']) {
+            const video = document.createElement('video');
+            video.muted = options['muted'];
+            video.defaultMuted = options['muted'];
+            video.autoplay = options['autoplay'];
+            video.preload = 'metadata';
+            video.controls = !(options['controls']);
+            video.loop = options['loop'];
+            video.style.width = '100%';
+            video.style.height = '90vh';
+            // Store selected options as part of the src URL
+            const autoplay = video.autoplay ? '?autoplay=1&mute=1' : '?autoplay=0';
+            const controls = video.controls ? '&controls=1' : '&controls=0';
+            const loop = video.loop ? '&loop=1' : '&loop=0';
+            const source = url + autoplay + controls + loop;
+            video.src = source;
+            // Set to 0.01 for canvas thumbnail generation, otherwise thumbnail is black
+            video.currentTime = 0.01;
+            let thumbnail = document.createElement('canvas');
+            const frame = document.createElement('iframe');
+            frame.src = source;
+            frame.srcdoc = video.outerHTML;
+            frame.frameBorder = '0';
+            frame.allowFullscreen = true;
+            frame.contentEditable = 'false';
```

Comment:
Well since the iframes are inserted as previews, maybe it was required. I am basically copying the getWrappedIframe(src) function which was there for embedding external source videos and it explicitly sets contenteditable=false. 

I supposed there was a reason for that, but I'm not sure what I should be looking for if I remove it.